### ISSUE TITLE
Require PHP 7.0

### ISSUE THREAD

**Issue by Al2Klimov** [Icinga/icingaweb2#4007]
Right now [we require 5.6](https://icinga.com/docs/icingaweb2/latest/CHANGELOG/#raised-php-version-dependency), but at the same time all of [the supported distros](https://icinga.com/subscription/support-details/) include 7.0 (RHEL: SCL, SLES: SDK).

We should take the opportunity of our next major version and make our lives a bit easier.

**Comment by Thomas-Gelf** [comment]
You can add Debian (since Stretch) and Ubuntu (since Xenial) to this list. RHEL and CentOS 7 without SCL are still on 5.6. While our CentOS packages depend on a (currently outdated and unsupported) SCL version, again and again we meet systems still running Icinga Web 2 via 5.6. This is the case where users customized their web server configuration long time ago. Upgrading our packages then didn't adjust their configuration. So while they're running our latest packages, Icinga Web 2 isn't running the way one might expect.

Many enterprises are doing so, as they tweak the Web Server for various reasons:

- Kerberos/SSO Authentication
- Running Icinga Web 2 in the DocumentRoot of a dedicated Domain
- Combining our default configuration with settings for third-party settings

Said all this, you can expect this change to lead to trouble for more than just a very few installations. Sure, sooner or later we have to go this step. Just, I'm not a big fan of causing trouble when there is no urgency to do so.

Personally, while requiring 7.1 or 7.3 would make some things easier, I do not see the big advantage of requiring PHP 7.0. I'm a big fan of PHP 7, as it brings an enormous performance boost. But for many of the new features 7.0 is not good enough. Once using some shiny "new" features, you'll very soon realize that 7.1 or 7.3 would be required to really benefit from what you gained.

Long story short: please explain your motivation for this request. What's your personal problem with 5.6, where do you feel restricted, where does it block your current tasks?

Thanks,
Thomas

**Comment by Al2Klimov** [comment]
First of all – and IMAO the most important of all: Sure, 7.x provides more features compared to 5.6 than just 7.0. But doing (let's say) a little bit and requiring (let's say) just 7.0 is already much more than just excusing oneself.

By the way I'd prefer a somewhat smoother transition than just (let's say) 5.6 to 7.3 not to get a bullet in my head from a user of the monitoring forum 😉. I'd even suggest @lippserd not to raise the PHP version requirement by more than one step by IW2 version.

Sure, we ~can~ *could* (!) add some older distros, but what for? This list is IMAO for removing, but not adding old ones. Or would *you* like to support Win2000 as a platform? (Have fun!)

What do I miss? Very good question, please read on. I (as a mod dev) didn't even waste my time for thinking about that. I could miss a lot, but no one will require 7.0 just because I miss it. E.g. I (as a mod dev) actually missed #2741, but it wasn't added. (Hello @lippserd!) Instead we'll (hopefully) require 7.0 *because we can* for the reasons I already explained:

1. We officially support just distros A, B and C, versions X, Y and Z.
2. The above ones have PHP 7.
3. If you as an operator would like to run the latest IW2 and get support from us (paid or in the monitoring forum), you have to meet our requirements which is pretty easy. We won't roast extra sausages for anyone. Period.

![](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQt9HPRXbjSbWH7pgYeai-lpVoV7ANrorvwttvtqRVgOytQQtRUQA&s)

If it's not that easy *for you*, feel free to ask @LordHepipud whether @Netways can help you. You'll get a positive answer – that guy sells everything that's not on top of the tree as fast as he has counted to three 😉.

![](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSEZBVpodSQZ88hmZ7-Mcx224hduOO6wE1kOKvVRz5WLWEifaN8vw&s)

Best,
AK

**Comment by Thomas-Gelf** [comment]
> What do I miss?

The point.

**Comment by dnsmichi** [comment]
I agree with @Thomas-Gelf here.

**Comment by nilmerg** [comment]
Let's discuss this on a later point. Thanks.

**Comment by Al2Klimov** [comment]
Superseded by #4397.
